Procfile

```
@@ -1,4 +1,4 @@
-web: WEBPACK_PORT=${WEBPACK_PORT:-3035} bundle exec rackup config.ru --port ${PORT:-3000} --host ${HOST:-localhost}
+web: WEBPACK_PORT=${WEBPACK_PORT:-3035} bundle exec rackup config.ru --port ${PORT:-3000} --host ${FOREMAN_HOST:-localhost}
 worker: bundle exec good_job start
 mailcatcher: mailcatcher -f
 js: WEBPACK_PORT=${WEBPACK_PORT:-3035} yarn webpack serve
```

per our comment about `yarn webpack --watch`, would it make sense to interpolate `--watch` in conditionally here? or was that going to cause HTTPS issues?

---

> per our comment about `yarn webpack --watch`, would it make sense to interpolate `--watch` in conditionally here? or was that going to cause HTTPS issues?

Yeah, I thought about that, but it wasn't immediately clear to me how that would look as far as syntax. If you know off-hand, happy to add it.

Basically, with the workflow as implemented here currently, the dev server would be run despite not actually requesting from it. It's not totally useless, since we also rely on it to do rebuilds on file changes, but the server component of it is unnecessary.

---

Based on .... https://unix.stackexchange.com/a/316462

Maybe
```suggestion
js: WEBPACK_PORT=${WEBPACK_PORT:-3035} yarn webpack serve $([ -n "$HTTPS" ] && echo --watch) 
```

---

So I think we want to switch between `webpack serve` for default dev-server, and `webpack --watch` for HTTPS. I guess we could probably do something similar with the inverse to `echo serve` ?

---

ah I see, like this then?
```suggestion
js: WEBPACK_PORT=${WEBPACK_PORT:-3035} yarn webpack $([ -n "$HTTPS" ] && echo "--watch" || echo "serve") 
```

---

That seems to work!

```
$ make run
13:23:11 js.1          | started with pid 97097
13:23:14 js.1          | yarn run v1.22.17
13:23:14 js.1          | $ /identity-idp/node_modules/.bin/webpack serve
```

```
$ HOST=0.0.0.0 make run-https
13:23:52 js.1          | started with pid 97216
13:23:53 js.1          | yarn run v1.22.17
13:23:53 js.1          | $ /identity-idp/node_modules/.bin/webpack --watch
```